lib/websocket.js

```diff
@@ -188,8 +190,19 @@ function update() {
               treatment.x = timestamp.getTime();
               return treatment;
           });
-          // all done, do loadData
-          loadData( );
+
+          profiles.list(function (err, results) {
+              // There should be only one document in the profile collection with a DIA.  If there are multiple, use the last one.
+              results.forEach(function(element, index, array) {
+                  if (element) {
+                    if (element.dia) {
```

Comment:
Lines 197-198 could be simplified to `if (element && element.dia) {`, correct?

Also, consider a standard `for` loop that goes backward from the end of `results`, and breaks when it finds `element.dia`.